recipe/build.sh

```diff
@@ -6,32 +6,63 @@ set -euxo pipefail
 
 _ARCH=${target_platform#*-} # keep chunk after dash (e.g. '64' in 'win-64')
 
+if [[ "${_ARCH}" == "64" ]]; then
+  _CL_MACHINE=x64
+elif [[ "${_ARCH}" == "arm64" ]]; then
+  _CL_MACHINE=ARM64
+fi
+
 # Build resources file
 test -f resources.rc && rm -f resources.rc
 echo "#include \"winuser.h\""      > resources.rc
 echo "1 RT_MANIFEST manifest.xml" >> resources.rc
 test -f resources-${_ARCH}.res && rm -f resources-${_ARCH}.res
-windres --input resources.rc --output resources-${_ARCH}.res --output-format=coff
+
+if [[ "${c_compiler}" == "gcc" ]]; then
+  ${WINDRES:-windres} --input resources.rc --output resources-${_ARCH}.res --output-format=coff -v
+else
+  which rc.exe
+  rc.exe resources.rc
+  mv resources.res resources-${_ARCH}.res
+fi
+
+ls -alh .
 
 # Compile launchers
 for _TYPE in cli gui; do
   if [[ ${_TYPE} == cli ]]; then
-    CPPFLAGS=
-    LDFLAGS=
+    if [[ "${c_compiler}" == "gcc" ]]; then
+      CPPFLAGS=
+      LDFLAGS=
+    else
+      CPPFLAGS=
+      LDFLAGS="-SUBSYSTEM:CONSOLE"
+    fi
   else
-    CPPFLAGS="-D_WINDOWS -mwindows"
-    LDFLAGS="-mwindows"
+    if [[ "${c_compiler}" == "gcc" ]]; then
+      CPPFLAGS="-D_WINDOWS -mwindows"
+      LDFLAGS="-mwindows"
+    else
+      CPPFLAGS="-D_WINDOWS"
+      LDFLAGS="-SUBSYSTEM:WINDOWS"
+    fi
   fi
 
-  # You *could* use MSVC 2008 here, but you'd end up with much larger (~230k) executables.
-  # cl.exe -opt:nowin98 -D NDEBUG -D "GUI=0" -D "WIN32_LEAN_AND_MEAN" -ZI -Gy -MT -MERGE launcher.c -Os -link -MACHINE:x64 -SUBSYSTEM:CONSOLE version.lib advapi32.lib shell32.lib
-  ${BUILD_PREFIX}/Library/mingw-w64/bin/gcc \
-    -O2 -DSCRIPT_WRAPPER -DUNICODE -D_UNICODE -DMINGW_HAS_SECURE_API -DMAXINT=INT_MAX ${CPPFLAGS} \
-    ${SRC_DIR}/launcher.c -c -o ${_TYPE}-${_ARCH}.o
+  # An executable with mingw and `static-libgcc` is 42K
+  # An executable with vcruntime statically linked (-MT) in is 920K
+  # An executable with vcruntime dynamically linked (-MD) in is 84K
+  # For arm64, since we don't have mingw, we are going to use -MD
+  #  to reduce the binary size. Let's hope that all arm64 machines
+  #  have vcruntime140.dll installed.
```

Comment:
@finnagin and @zooba, is this correct or should we go with `-MT`? The executables built here need to able to run without the conda supplied `vcurntime140.dll`

Reply:
Hmm, we don't need to run standalone. So let's use MD

Reply:
We do need them to run from `Scripts` which doesn't have a `vcruntime140.dll`.

Reply:
So do we need `-MD` or `-MT`? I think we are currently using `MD` but your last message hints at needing `MT` as far as I can see? 🤔

Reply:
I would use `-MT` for safety for launchers - it's too likely to end up running from a place where it can't find a suitable runtime.

Using `-MD` matter more for Python itself, because extension modules will also need to load it, and they'll want to share state. The launcher (I assume) is only going to create a new process, rather than dynamically loading DLLs, so it won't be a huge impact.